tests/ctl/core/test_api.py

```diff
@@ -571,15 +597,62 @@ async def test_system_create(
         )
 
         assert result.status_code == HTTP_201_CREATED
-        assert result.json()["name"] == "Test System"
-        assert result.json()["cookies"] == [
+        json_results = result.json()
+        assert json_results["name"] == "Test System"
+        assert json_results["cookies"] == [
             {"name": "essential_cookie", "path": "/", "domain": "example.com"}
         ]
-        assert result.json()["privacy_declarations"][0]["cookies"] == [
+        assert json_results["privacy_declarations"][0]["cookies"] == [
             {"name": "essential_cookie", "path": "/", "domain": "example.com"}
         ]
-        assert result.json()["privacy_declarations"][1]["cookies"] == []
-        assert len(result.json()["privacy_declarations"]) == 2
+        assert json_results["privacy_declarations"][1]["cookies"] == []
+        assert len(json_results["privacy_declarations"]) == 2
+
+        assert json_results["vendor_id"] == "test_vendor"
```

Comment:
I think there's a cleaner way to test these

maybe something like:

```python
for key, value in system.items():
    assert json_results[key] = value
```

Reply:
sure, I can compare the json response to the system values in a programmatic way, but I'd like to keep the system model assertions.